### PR TITLE
Remove publish steps from release pipeline

### PR DESCRIPTION
**Pull request by bachuv**
This PR updates the release pipeline to remove the publish steps. The changes include:

- updates to each projects .csproj to add the following code
```
<ItemGroup Condition="'$(Configuration)'=='Release'">
    <Content Include="..\..\_manifest\**">
      <Pack>true</Pack>
      <PackagePath>content/SBOM</PackagePath>
    </Content>
</ItemGroup>
```
- delete each project's .nuspec file as it's not needed anymore
- delete the `dotnet publish` step in the release pipeline .yml file

### CODE REVIEW

**Review by cgillum — APPROVED** [review]
I love how much simpler this is now! :)

File: src/DurableTask.Core/DurableTask.Core.csproj [hunk 1]
```diff
@@ -35,9 +35,11 @@
...
    
-   <PropertyGroup Condition="'$(Configuration)'=='Release'">
-    <NuspecFile>DurableTask.Core.nuspec</NuspecFile>
-    <NuspecProperties>configuration=$(Configuration);targetFramework=$(TargetFramework);version=$(Version)</NuspecProperties>
-  </PropertyGroup>
+  <ItemGroup Condition="'$(Configuration)'=='Release'">
+    <Content Include="..\..\_manifest\**">
+      <Pack>true</Pack>
+	  <PackagePath>content/SBOM</PackagePath>
+	</Content>
```
Comment (bachuv) [inline src/DurableTask.Core/DurableTask.Core.csproj:42]: This indentation looks fine locally and I'm not sure why it's off here. I'm looking into it.

Comment (cgillum) [reply src/DurableTask.Core/DurableTask.Core.csproj:42]: There's a mix of tabs and spaces. You can tell when you try to highlight the whitespace.

Comment (bachuv) [reply src/DurableTask.Core/DurableTask.Core.csproj]: I see, thanks! I fixed the indentation and I'm running the pipeline again just to confirm that it succeeds. Once it succeeds, I'll merge the changes.

**Pull request merged**